common/sketch/src/test/scala/org/apache/spark/util/sketch/BloomFilterSuite.scala

```diff
@@ -0,0 +1,109 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.util.sketch
+
+import scala.reflect.ClassTag
+import scala.util.Random
+
+import org.scalatest.FunSuite // scalastyle:ignore funsuite
+
+class BloomFilterSuite extends FunSuite { // scalastyle:ignore funsuite
+
+  def testAccuracy[T: ClassTag](typeName: String, numItems: Int)(itemGen: Random => T): Unit = {
+    test(s"accuracy - $typeName") {
+      // use a fixed seed to make the test predictable.
+      val r = new Random(37)
+      val fpp = 0.05
+      val numInsertion = numItems / 10
+
+      val allItems = Array.fill(numItems)(itemGen(r))
+
+      val filter = BloomFilter.create(numInsertion, fpp)
+
+      // insert first `numInsertion` items.
+      var i = 0
+      while (i < numInsertion) {
+        filter.put(allItems(i))
+        i += 1
+      }
```

Comment:
I think it's OK to use `allItems.take(numInsertion).foreach(filter.put)` for simplicity if this code path doesn't slow down test execution too much.